src/libcollections/vec_deque.rs

```diff
@@ -600,8 +713,17 @@ impl<T> VecDeque<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn is_empty(&self) -> bool { self.len() == 0 }
 
-    /// Creates a draining iterator that clears the `VecDeque` and iterates over
-    /// the removed items from start to end.
+    /// Create a draining iterator that removes the specified range in the
+    /// `VecDeque` and yields the removed items from start to end. The element
+    /// range is removed even if the iterator is not consumed until the end.
+    ///
+    /// Note: It is unspecified how many elements are removed from the deque,
+    /// if the `Drain` value is leaked.
```

Comment:
Even in those cases, we can't guarantee that the values we were going to drain are actually dropped. All we can guarantee is that the vecdeque will contain the correct values in the end.

I'm not sure that making the leaking guarantee a bit stronger is that valuable.